ts/src/test/test.ts

```diff
@@ -723,8 +715,14 @@ export default class testMainClass extends baseMainTestClass {
         if (this.sandbox || getExchangeProp (exchange, 'sandbox')) {
             exchange.setSandboxMode (true);
         }
-        await this.loadExchange (exchange);
-        await this.testExchange (exchange, symbol);
+        try {
+            await this.loadExchange (exchange);
+            await this.testExchange (exchange, symbol);
+            await close (exchange);
+        } catch (e) {
+            await close (exchange); // close for async-python
+            throw e;
+        }
```

Comment:
this above try/catch lines were needed only for async-python, to trigger exchange-closure before throwing an exception